go/libraries/doltcore/sqle/dolt_diff_summary_table_function.go

```diff
@@ -0,0 +1,503 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"fmt"
+	"io"
+	"math"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/doltcore/diff"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+	"github.com/dolthub/dolt/go/store/atomicerr"
+)
+
+var _ sql.TableFunction = (*DiffSummaryTableFunction)(nil)
+
+type DiffSummaryTableFunction struct {
+	ctx *sql.Context
+
+	fromCommitExpr sql.Expression
+	toCommitExpr   sql.Expression
+	tableNameExpr  sql.Expression
+	database       sql.Database
+}
+
+var diffSummaryTableSchema = sql.Schema{
+	&sql.Column{Name: "table_name", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "rows_unmodified", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "rows_added", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "rows_deleted", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "rows_modified", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_added", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_deleted", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "cells_modified", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "old_row_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "new_row_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "old_cell_count", Type: sql.Int64, Nullable: true},
+	&sql.Column{Name: "new_cell_count", Type: sql.Int64, Nullable: true},
+}
+
+// NewInstance implements the TableFunction interface
```

Comment:
(super minor) technically, not part of the TableFunction interface. Would be nice to put the standard "Create a new instance of ..."